### PR TITLE
Rule controller tests

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
This PR Introduces integration tests for oathkeeper-k8s-controller that run a "happy-path" scenario against a K8s environment.
There is a Make target provided for running it locally using `KIND` environment: `make test-integration`
You can also launch the tests against a real K8s cluster.

**Note: It will be rewritten to stretchr/testify assertion lib, see: #10 **

### CODE REVIEW

File: tests/integration/integration_test.go.bak [hunk 1]
```diff
@@ -0,0 +1,316 @@
+package integration
```
Comment (aeneasr) [inline tests/integration/integration_test.go.bak]: I think this file can be deleted?

File: tests/integration/validation_test.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
+package integration_test
```
Comment (strekm) [inline tests/integration/validation_test.go]: I would suggest renaming this file. ATM it suggests it's a test.

File: tests/integration/retry_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package integration_test
```
Comment (strekm) [inline tests/integration/retry_test.go]: I would suggest renaming this file. ATM it suggests it's a test.

File: tests/integration/rules_test.go [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+	//Delete namespace
+	err := deleteNamespace(namespaceName, k8sClient)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = Describe("Oathkeeper controller", func() {
+
+	Context("in happy path scenario", func() {
+
+		It("should", func() {
```
Comment (jakkab) [inline tests/integration/rules_test.go]: how 'bout:
```
var _ = Describe("Oathkeeper controller", func() {

	It("should represent all in-cluster oathkeeper rules as a single entry in a Configmap", func() {

		By("creating a valid Configmap upon first reconciliation")
```

etc.

File: tests/integration/rules_test.go [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+
+			By("create a valid ConfigMap from a single Rule")
+			//Given
+			rule, err := getRule(getRule1Json())
+			Expect(err).To(BeNil())
+			Expect(rule).ToNot(BeNil())
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply tests/integration/rules_test.go]: Done.

File: tests/integration/rules_test.go [hunk 6]
```diff
@@ -0,0 +1,467 @@
...
+func ensureRule(rule *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+
+	//Create
+	_, createErr := k8sDynamicClient.Resource(ruleResource).Namespace(namespaceName).Create(rule, metav1.CreateOptions{})
+	Expect(createErr).To(BeNil())
+
+	//Wait until Rule can be read from the cluster.
+	workerFunc := func() (interface{}, error) {
+
+		rule, err := k8sDynamicClient.Resource(ruleResource).Namespace(namespaceName).Get(rule.GetName(), metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply tests/integration/rules_test.go]: You are right, this is just an idiomatic template burned into my brain. We can return immediately.

File: tests/integration/rules_test.go (hunk 4 above)
Comment (Tomasz-Smelcerz-SAP) [reply tests/integration/rules_test.go]: After thinking about it, I chose this version:
```
Oathkeeper controller
  should manage rules as ConfigMap entries
    in happy path scenario
```

**Comment by aeneasr** [comment]
Awesome!

**Pull request merged**